### PR TITLE
Add endpoint to list users

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
- Adds new endpoint to list users - `/api/v2/users`
- Adds a new `lastSeen` field to the user crd spec
- Updates `UserSaver` middleware to update `lastSeen` date

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #6563

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Add an endpoint to list users and extend the user object to allow checking the last seen date.
```

### CODE REVIEW

File: pkg/controller/master-controller-manager/user-synchronizer/controller_test.go [hunk 1]
```diff
@@ -104,6 +104,9 @@ func TestReconcile(t *testing.T) {
 					t.Fatalf("failed to get user: %v", err)
 				}
 			} else {
+				seedUser = sanitize(seedUser)
```
Comment (floreks) [inline pkg/controller/master-controller-manager/user-synchronizer/controller_test.go:107]: @xrstf could you PTAL? I couldn't force the controller to make an equal copy of the `lastSeen` field and it was always failing due to differences in the internal `location` field. If you've got a better idea of how to make this work, I will change this.

File: pkg/handler/middleware/middleware.go [hunk 2]
```diff
@@ -184,6 +188,15 @@ func UserSaver(userProvider provider.UserProvider) endpoint.Middleware {
...
 				}
 			}
+
+			user.Spec.LastSeen = &[]metav1.Time{metav1.NewTime(Now().UTC())}[0]
+			user, err = userProvider.UpdateUser(user)
+			if err != nil {
+				return nil, common.KubernetesErrorToHTTPError(err)
+			}
+
+			fmt.Println(user.Spec)
```
Comment (floreks) [inline pkg/handler/middleware/middleware.go]: Remove this :slightly_smiling_face:

File: pkg/handler/middleware/middleware.go [hunk 3]
```diff
@@ -184,6 +188,15 @@ func UserSaver(userProvider provider.UserProvider) endpoint.Middleware {
 					}
 				}
 			}
+
+			user.Spec.LastSeen = &[]metav1.Time{metav1.NewTime(Now().UTC())}[0]
+			user, err = userProvider.UpdateUser(user)
+			if err != nil {
```
Comment (floreks) [inline pkg/handler/middleware/middleware.go:200]: Do we have a check for the "object has been modified" error as it is not critical here? We can assume that another call just updated the time.

Comment (floreks) [reply pkg/handler/middleware/middleware.go:200]: Found and updated

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**